translator-deepl/src/main/java/org/xwiki/contrib/translator/deepl/DeeplTranslator.java

```diff
@@ -108,6 +147,94 @@ public Usage getUsage() throws DeepLException, InterruptedException
         return translator.getUsage();
     }
 
+    @Override
+    public List<LocalePairs> getGlossaryLocalePairs() throws TranslatorException
+    {
+        Translator translator = getTranslator();
+        try {
+            return translator.getGlossaryLanguages().stream()
+                .map(item -> new LocalePairs(item.getSourceLanguage(), item.getTargetLanguage()))
+                .collect(Collectors.toList());
+        } catch (InterruptedException e) {
+            logger.debug("Error when getting glossary languages [{}]", e.getMessage(), e);
+            throw new TranslatorException("Interrupt exception getting glossary languages", e);
+        } catch (DeepLException e) {
+            logger.debug("Error when getting glossary languages [{}]", e.getMessage(), e);
+            throw new TranslatorException("DeepL exception when getting glossary languages", e);
+        }
+    }
+
+    @Override
+    public List<GlossaryInfo> getGlossaries() throws TranslatorException
+    {
+        Translator translator = getTranslator();
+        try {
+            String glossaryNamePrefix = getGlossaryNamePrefix();
+            return translator.listGlossaries()
+                .stream()
+                .filter(entry -> entry.getName().startsWith(glossaryNamePrefix))
+                .map(item -> new GlossaryInfo(item.getGlossaryId(), item.getName(), item.isReady(),
+                    item.getSourceLang(), item.getTargetLang(), item.getEntryCount()))
+                .collect(Collectors.toList());
+        } catch (InterruptedException e) {
+            logger.debug("Error when getting glossaries [{}]", e.getMessage(), e);
+            throw new TranslatorException("Interrupt exception when getting glossaries", e);
+        } catch (DeepLException e) {
+            logger.debug("Error when getting glossaries [{}]", e.getMessage(), e);
+            throw new TranslatorException("DeepL exception when getting glossaries", e);
+        }
+    }
+
+    @Override
+    public Map<String, String> getGlossaryEntryDetails(String id) throws TranslatorException
```

Comment:
Might be renamed into `getGlossaryEntries` instead?

Reply:
Fixed by 66e6e01a77520d038148890cec8c57980b121f65